test/pylib/suite/cql_approval.py

```diff
@@ -8,200 +8,207 @@
-                    set_summary("""returned non-zero return status.\n
-Check test log at {}.""".format(self.log_filename))
-                elif not os.path.isfile(self.tmpfile):
-                    set_summary("failed: no output file")
-                elif not os.path.isfile(self.result):
-                    set_summary("failed: no result file")
-                    self.is_new = True
-                else:
-                    self.is_equal_result = filecmp.cmp(self.result, self.tmpfile)
-                    if not self.is_equal_result:
-                        self.unidiff = format_unidiff(str(self.result), self.tmpfile)
-                        set_summary("failed: test output does not match expected result")
-                        assert self.unidiff is not None
-                        logger.info("\n{}".format(palette.nocolor(self.unidiff)))
-                    else:
-                        self.success = True
-                        set_summary("succeeded")
-            except Exception as e:
-                # Server log bloats the output if we produce it in all
-                # cases. So only grab it when it's relevant:
-                # 1) failed pre-check, e.g. start failure
-                # 2) failed test execution.
-                if not self.is_executed_ok:
-                    self.server_log = cluster.read_server_log()
-                    self.server_log_filename = cluster.server_log_filename()
-                    if not self.is_before_test_ok:
-                        set_summary("pre-check failed: {}".format(e))
-                        print("Test {} {}".format(self.name, self.summary))
-                        print("Server log  of the first server:\n{}".format(self.server_log))
-                        # Don't try to continue if the cluster is broken
-                        raise
-                set_summary("failed: {}".format(e))
-            finally:
-                if os.path.exists(self.tmpfile):
-                    if self.is_executed_ok and (self.is_new or not self.is_equal_result):
-                        # Move the .reject file close to the .result file
-                        # so that it's easy to analyze the diff or overwrite .result
-                        # with .reject.
-                        shutil.move(self.tmpfile, self.reject)
-                    else:
-                        pathlib.Path(self.tmpfile).unlink()
+        return f"*{CQL_TEST_SUFFIX}"
+
+
+class CqlFile(Module):
+    obj = None
+
+    def funcnamefilter(self, name: str) -> bool:
+        return True
+
+    def collect(self) -> Iterable[CqlTest]:
+        assert self.ihook.pytest_pycollect_makeitem(
+            collector=self,
+            name=self.name,
+            obj=self.obj,
+        ) is None, ".cql file will be not collected as a Python module"
+        yield CqlTest.from_parent(parent=self, name=self.name.removesuffix(CQL_TEST_SUFFIX))
+
 
-        return self
+class CqlTest(pytest.Item):
```

Comment:
I would better move this outside the suite file. Suits will be deleted in the nearest future. I have a similar custom collector for C++ files, and it's located under `test/pylib/cpp`. I guess it's a better location for custom collectors. Moreover, we already have `cql_approval` under `pylib`.

Reply:
Moved to `pylib/cql_repl.py`